when cleaning up txns, actually do that

tbg opened the pull request:
current implementations could wind up pushing abandoned transactions
into the future when touched through (*Store).resolveWriteIntentError.

this is one of the things that tripped me up during today's demo: #3031 
abandoned the transaction, so the intent wasn't removed. running the
transaction again was impossible since the pending intent would always
be pushed into the future, and #2861 kicked in and gave me a
`ReadWithinUncertaintyIntervalError` - so in fact on the console there was
no way to get rid of the intent.

With this change, at least a push after 10 seconds will actually remove the
intent, not push it into the future.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3032)

<!-- Reviewable:end -->

tamird commented on storage/replica_test.go:
```diff
@@ -2171,15 +2171,18 @@ func TestPushTxnHeartbeatTimeout(t *testing.T) {
 
 		if test.expSuccess != (err == nil) {
-			t.Errorf("expected success on trial %d? %t; got err %s", i, test.expSuccess, err)
+			t.Errorf("%d: expected_success=%t; got err %s", i, test.expSuccess, err)
```
either leave it as "expected success" or use "expSuccess"

tschottdorf replied:
done.

tamird commented:
LGTM

tbg committed:
when cleaning up txns, actually do that

current implementations could wind up pushing abandoned transactions
into the future when touched through (*Store).resolveWriteIntentError.
